src/libs/actions/Task.js

```diff
@@ -0,0 +1,102 @@
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+
+/**
+ * Assign a task to a user
+ * @param {String} parentReportActionID
+ * @param {String} parentReportID
+ * @param {String} taskReportID
+ * @param {String} name
+ * @param {String} description
+ * @param {String} assignee
+ * @param {String} ownerEmail
+ *
+ */
+
+function assignTask(parentReportActionID, parentReportID, taskReportID, name, description, assignee, assigneeChatReportID, ownerEmail) {
+    let parentReport;
+
+    // If there isn't a parentReportId provided and assignee is given, we need to optimistically create a new parent chat report
+    // between the task creator and the assignee.
+    if (!parentReportID && assignee) {
+        parentReport = ReportUtils.getChatByParticipants([assignee]);
+    }
+
+    if (!parentReportID && !parentReport && assignee) {
+        parentReport = ReportUtils.buildOptimisticChatReport([assignee]);
+    }
+
+    // If parentReport is defined, use its reportID, otherwise, use the provided parentReportID
+    const finalParentReportID = parentReport ? parentReport.reportID : parentReportID;
+
```

Comment:
This flow can be pretty confusing just because there is so much going on - throwing down some thoughts to see if we can simplify this flow a bit

// Is there an assignee? 
YES - Let's check if we need to message the assignee separately (Is task shared in a group chat / room / DM the assignee)
NO - Great, no need to create a seperate chat report

// Is the task shared somewhere?
YES - Cool, the place where the task is shared will always exist
NO - By default, the parentReportID will be the chat between the assignee and creator. 

Task must have either an assignee OR a place where it is shared.  Doesn't need both - but needs one or the other. If the task only has an assignee, we infer that it is shared in the DM.

// Do we need to message the assignee separately?
YES - Grab the chat between assignee and creator (if needed create optimistically) and then send them the task
NO - Nice

// ALWAYS
We will always send a message to the parent chat report